lib/cocoapods/target/build_settings.rb

```diff
@@ -1246,6 +1292,17 @@ def user_target_xcconfig_values_by_consumer_by_key
           merged_xcconfigs(user_target_xcconfig_values_by_consumer_by_key, :user_target_xcconfig)
         end
 
+        # @see BuildSettings#should_apply_xctunwrap_fix?
+        def should_apply_xctunwrap_fix?
+          target.library? && target.user_targets.any? do |ut|
+            next false unless Version.new(ut.deployment_target) < Version.new('12.2')
+            flags = (ut.resolved_build_setting('OTHER_LDFLAGS', :resolve_against_xcconfig => true) || {}).values.flatten.uniq
+            parsed_flags = Xcodeproj::Config::OtherLinkerFlagsParser.parse(flags)
+            all_frameworks = (parsed_flags[:frameworks] + parsed_flags[:weak_frameworks]).uniq
+            all_frameworks.include?('XCTest') || all_frameworks.include?('"XCTest"')
```

Comment:
why can't we check whether the build setting includes `XCTest` in it's own set of frameworks?

Reply:
you mean check if XCTest is present in any of the `pod_targets` inside the aggregate target?

Reply:
actually checking against the pod targets wouldn't necessarily fix this target. We basically do what the user should do for their own target automatically here. It is possible the user target is a framework or library that depends on 10 pods that don't use XCTest but the user target itself does and would need this fix.

Reply:
but they'd have that issue without cocoapods as well, then?

Reply:
Thats correct they would. The above as it is currently is would fix it automatically for the user without them having to manually apply their fix on their own target.

Reply:
Are you suggesting to _also_ check if any of the `pod_targets` within the aggregate have `XCTest` then to apply the fix or _replace_ it?

Reply:
Alternatively we can skip this check entirely for aggregate targets and let uses apply the Apple fix manually for their own targets. Happy to remove it entirely.

Reply:
> Alternatively we can skip this check entirely for aggregate targets and let uses apply the Apple fix manually for their own targets. Happy to remove it entirely.


I think I'd be in favor of that

Reply:
will remove.

Reply:
removed.